### PR TITLE
[ONNX] Update expand_as for dynamic shape

### PR DESCRIPTION
**Pull request by neginraoof**
Update expand_as for dynamic shape

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -470,6 +470,12 @@ def expand(g, self, size, implicit):
 
 
 def expand_as(g, self, other):
+    self_t = sym_help._maybe_get_const(self, "t")
+    if isinstance(self_t, torch.Tensor):
+        self_t = self_t.to(torch.double)
+        if torch.equal(self_t, torch.full_like(self_t, torch.mean(self_t))):
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:476]: can this support non-scalar cases? e.g. `x[:, x.size(0):, :] = torch.tensor([1,2,3])` where `x.shape == [2, 5, 3]`.

Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:476]: We are only supporting cases where right hand side is a constant tensor with uniform value.

Comment (BowenBao) [reply torch/onnx/symbolic_opset9.py:476]: Do you mean the above is not supported, or it will not fall under this code path?

**Comment by BowenBao** [comment]
CI failures of onnx tests are related

**Comment by neginraoof** [comment]
@BowenBao Tests should be skipped for lower opsets. Do you have any comments on the updates?

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: So, I was thinking to keep it out of the scope, but it should be fixed now.

**Review by BowenBao — APPROVED** [review]
LGTM, thanks!

**Pull request merged**